src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFunction = (a, b) => (String(a) > String(b) ? 1 : -1);
+    const actualCompareFunction =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFunction;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        if (actualCompareFunction(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
```

Comment:
The bubble sort algorithm is correctly implemented, but it's worth noting that bubble sort has a time complexity of O(n^2), which is not efficient for large datasets. For better performance, consider implementing a more efficient sorting algorithm like quicksort or mergesort in the future.